fastlane/spec/actions_specs/git_default_remote_branch.rb

```diff
@@ -0,0 +1,101 @@
+describe Fastlane do
+  describe Fastlane::FastFile do
+    directory = "fl_spec_default_remote_branch"
+    repository = "https://github.com/seanmcneil/Empty.git"
```

Comment:
Naah...not this way, if this repository will delete for some reason, Fastlane CI will break in `git clone` step.
I will recommend unit tests using mocks with fake/stub responses. 

This is a nice spec where you can see how to mock git commands using fake/stub responses
https://github.com/fastlane/fastlane/blob/master/fastlane/spec/actions_specs/push_to_git_remote_spec.rb

Please let me know if need more help! Happy to help :innocent:

Reply:
I completely agree and had this specific item in mind with my "I'm not sure I am doing this right comment". I am still working on this item and it may be a bit before I can resume work and correct it.

Reply:
I updated the tests based on the push spec you linked. My use case was a little different, but I hope I implemented it correctly. Please let me know if there is anything I missed.